src/shared/dissect-image.c

```diff
@@ -1051,6 +1052,12 @@ static int mount_partition(
                 if (!strextend_with_separator(&options, ",", m->mount_options, NULL))
                         return -ENOMEM;
 
+        if (FLAGS_SET(flags, DISSECT_IMAGE_MKDIR)) {
+                r = mkdir_p(p, 0755);
```

Comment:
in src/core/mount.c the target directory is created with the selinux version (mkdir_p_label) - is there any specific logic to when one is used vs the other?

Reply:
basically: unless it's really really obvious that we want selinux labelling we don't do it. and wait until some selinux people complain and then add it. Thing is, we don't maintain the policy, and we don't really know if the want to label something in some specific way or are fine with default labels.

Moreover, this stuff is relative to some image, and that image if it has selinux has likely a different policy applied, hence if we really tried to label things correctly here, we'd have to load the policy from the image, and selinux has no recursive policy loading/namespacing, hence we can't do it even if we would want to...

Reply:
Right, I've been pestering our SELinux folks for quite some time about namespaced policies, for this and other reasons. My plan is to keep annoying them until they implement it just to shut me up :-D
More seriously, this one is probably going to be one of the use cases where we need the labelling to happen. I'll ping them and follow up personally if needed, once we figure it out.

Reply:
@bluca note that this is only a fallback anyway, if the mount points to overmount are missing so far. Any properly build image should not need this, and the mkdir_p() should be a NOP.

I mostly added this, so that I can use systemd-repart to generate an image with initialized file systems but zero files in them, and still mount it with "systemd-dissect --mount". i.e. I wanted to be able to take an image consistent of a few partitions fresh out of "mkfs" and friends, mount it with `systemd-dissect --mount` and then rsync a file tree into it, to populate it. But given that mkfs leaves entirely empty fs by default, the dirs to mount over are not going to exist either in this case, so I figured, let's just create them — if they are missing.

A proper, ppoulated, selinux enabled image doesn't need any of this: the mkdir_p() should be a NOP there, because the mount point dirs would already exist.

Hence, I think it's totally fine not to do the selinux dance here: it's a fallback that is only useful in very specific cases, where you start from scratch. in the normal case it should not be needed, and if you have a problem with the labels it puts on stuff then the right fix is probably to create and label the mountpoint dirs beforehand properly, so that `systemd-dissect --mount` doesn't have to anymore.